ocw/lib/gce.py

```diff
@@ -207,3 +209,112 @@ def cleanup_all(self):
+            keys += [k for k in service_resp['keys']]
+        return keys
+
+    @filterService(name='vaultopenqa')
+    def parser_service_account_list(self, iam_service):
+        ''' Returns a list of the emails of the Service Accounts
+
+        The request is using the resource name of the project associated with
+        the service accounts. This returns a list of dict items, which are
+        also not vault related.
+        The objects returned in chunks which contain a `nextPageToken`
+        to the next page. The `parser_service_account_list` parses each page
+        and returns the full list. Subsequently, the list is filtered to return
+        only emails correlated with vault
+
+        Parameters
+        ----------
+        iam_service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        '''
+        vault_emails = []
+        req = iam_service.projects().serviceAccounts().list(
+            name='projects/suse-sle-qa')
+
+        while True:
+            resp = req.execute()
+            vault_emails += [service_account['email']
+                             for service_account in resp.get('accounts', [])]
+
+            req = iam_service.projects().serviceAccounts().list_next(
+                previous_request=req, previous_response=resp)
+            if req is None:
+                break
+        return tuple(set(vault_emails))
+
+    def delete_vault_keys(self, iam_service, vault_keys):
+        ''' Deletes a list of vault keys
+
+        Expects _iam_service_ of type *googleapiclient.discovery.Resource*
+        and _vault_keys_ list of strings.
+
+        A short comment from the documentation. - Deleting a service account
+        key does not revoke short-lived credentials
+        that have been issued based on the service account key.
+
+        Parameters
+        ----------
+        iam_service : googleapiclient.discovery.Resource, required
+            The IAM Service Account instance
+        vault_keys : list, required
+            A list of keys to delete. This is retrieved by the _accounts_ of
+            the *parser_service_account_list* which is the email field.
+
+        Raises
+        ------
+        TypeError
+            This will raised when the _vault_keys_ will not match the expected pattern in the _name_
+        HttpError
+            Http Respond Errors
+        '''
+        if (len(vault_keys) < 1):
+            self.log_info("Nothing to delete")
+        else:
+            for vault_key in list(vault_keys):
+                if self.dry_run:
+                    self.log_warn(
+                        "Deletion of vault key {} skipped due to dry run mode",
+                        vault_key
+                    )
+                else:
+                    try:
+                        iam_service.projects().serviceAccounts().keys().delete(
+                            name=vault_keys).execute()
```

Comment:
```suggestion
                            name=vault_key).execute()
```

Reply:
*this change is ready for review*

Reply:
i havent started the writing of the testing yet. i am going to do it as a very last step. unless..? Do you find good reason to start with it now?